src/core/unit.c

```diff
@@ -5385,7 +5410,10 @@ static int unit_export_invocation_id(Unit *u) {
         if (sd_id128_is_null(u->invocation_id))
                 return 0;
 
-        p = strjoina("/run/systemd/units/invocation:", u->id);
+        r = unit_get_invocation_path(u, &p);
+        if (r < 0)
+                return log_unit_debug_errno(u, r, "Failed to get invocation path: %m");
```

Comment:
hmm, we'd likely log twice here now, since unit_get_invocation_path() already logs about one half of the errors. usually we try to avoid duplicate error logging: in this case it's not too important, given that both are just debug logging, but it would still be great if we'd just log once here.